xcmds/pex/pex.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2012-2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// pex builds a portable executable as a squashfs image.
+// It is intended to create files compatible with tinycore
+// tcz files.
+// This could have been a simple program but mksquashfs does not
+// preserve path information.
+// Yeah.
+//
+// Synopsis:
+//     pex [-d] -[output|o file]
+//
+// Description:
+//     pex makes portable executables in squashfs format compatible with
+//     tcz format. We don't build in the execution code, rather, we set it
+//     up so we can use the command itself.
+//
+// Options:
+//     debug|d: verbose
+//     output|o file: output file name (default /tmp/pex.tcz)
+//     test|t: run a test by loopback mounting the squashfs and using the first arg as a command to run in a chroot
```

Comment:
For testing, you should also have a `pex_test.go` file which can be run automatically by `go test`.